File: python/is_anagram.py
def is_anagram(s: str, t: str):
    char_set_s = [26]
    char_set_t = [26]

    for c in s:
        char_set_s[ord(c.lower()) - ord('a')] += 1
    
    for c in t:
        char_set_t[ord(c.lower()) - ord('a')] += 1
    
    for i in range(len(char_set_s)):
        s_count = char_set_s[i]
        if char_set_t[i] != s_count:
            return False

    return True

File: cpp/src/is_anagram.cpp
#include <string>
#include <vector>
using namespace std;

bool is_anagram(string s, string t) {
    vector<int> char_set_s;
    vector<int> char_set_t;

    for (char c : s) {
        char_set_s[c - 'A']++;
    }

    for (char c : t) {
        char_set_t[c - 'A']++;
    }

    for (int i = 0; i < char_set_s.size(); i++) {
        int count_s = char_set_s[i];

        if (char_set_t[i] != count_s) {
            return false;
        }
    }

    return true;
}